x-pack/plugins/security_solution/public/timelines/components/timeline/esql_tab_content/index.tsx

```diff
@@ -238,10 +241,11 @@ export const DiscoverTabContent: FC<DiscoverTabContentProps> = ({ timelineId })
         savedSearchAppState = getAppStateFromSavedSearch(localSavedSearch);
       }
 
-      const defaultDiscoverAppState = await getDefaultDiscoverAppState();
-
       const finalAppState =
-        savedSearchAppState?.appState ?? discoverAppState ?? defaultDiscoverAppState;
+        savedSearchAppState?.appState ??
+        stateContainer.appState.get() ??
+        discoverAppState ??
+        defaultDiscoverAppState;
```

Comment:
I know this matches existing behavior, but just wanted to make a UX note that dropping users off on an error state when initialing navigating to the ES|QL tab isn't the best experience. Here empty state === error state, so it logically makes sense, but isn't helpful:

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/6f393620-d4c6-4949-829c-22a09bf2ea07" />
</p> 

If not wanting a curated first time user experience (placeholder content with helptext for building esql queries or something), we could always default to what Discover does and pre-fill with a `Security Default Data View` limit-n query, so somewhat relevant security content is displayed by default if available:

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/b1986f2e-dcce-4adb-92c8-0512be8b31be" />
</p> 

---

No change necessary here @patrykkopycinski, just commenting around UX for the timeline folks.